### ISSUE TITLE
Make Home and End Buttons Work in Editor

### ISSUE THREAD

**Issue by JoelProminic** [Moonshine-IDE/Moonshine-IDE#1225]
Home should move the cursor to the top of the editor.  End should move to the cursor to the last line of the editor (I think the beginning of the line makes sense to avoid horizontal scroll issues if the last line is long, but if we are combining this with other shortcuts like selection, we may find that the end of the line is more useful).

**Comment by rat-moonshine** [comment]
When I looked into this today, I see the proposed behavior already available and working on both macOS and Windows.

I compared the behaviors are identical with other text editors, too. Following is the keyboard-short and their behaviors I found working in Moonshine-IDE editor:

Shortcuts | Behavior
--------------|--------------------
Home | Cursor horizontally jump to leftmost of the line
End | Cursor horizontally jump to rightmost of the line
Ctrl+Home | Scroll/moves cursor to beginning of the editor's 0th char
Ctrl+End | Scroll/moves cursor to end of the editor's line and to last character 

Please, check.

**Comment by JoelProminic** [comment]
I confirmed the shortcuts work on macOS Catalina.   I remember that Home and End didn't do anything when I originally reported this issue, so I'm not sure if this was implemented later, or if the shortcuts were broken when I tested this

The shortcuts match what I see in VS Code, and they seem to be the Windows standard.  I see some discussion about [heeding Fn + Arrow Keys](https://answers.microsoft.com/en-us/msoffice/forum/all/keyboard-shortcut-for-end-of-document/fa7f72a0-000a-4062-a9e6-ee32629edd34) on laptops, but it sounds like this may be mapped automatically.

I found [this page](https://support.apple.com/en-us/HT201236) on the standard macOS shortcuts.  There are many similar shortcuts here but the most comparable commands are CMD+Arrow Key. I confirmed that these also already work on macOS Catalina.

>Command–Up Arrow: Move the insertion point to the beginning of the document.
>Command–Down Arrow: Move the insertion point to the end of the document.
>Command–Left Arrow: Move the insertion point to the beginning of the current line.
>Command–Right Arrow: Move the insertion point to the end of the current line.

If we are satisfied with these commands, we can close this issue.  @rat-moonshine, did you confirm the shortcuts on Windows?
